x-pack/plugins/lens/public/editor_frame_service/editor_frame/editor_frame.tsx

```diff
@@ -118,56 +118,74 @@ export function EditorFrame(props: EditorFrameProps) {
   );
   const datasourceLayers = createDatasourceLayers(props.datasourceMap, state.datasourceStates);
 
-  const framePublicAPI: FramePublicAPI = {
-    datasourceLayers,
-    activeData,
-    dateRange,
-    query,
-    filters,
-    searchSessionId,
-    availablePalettes: props.palettes,
-
-    addNewLayer() {
-      const newLayerId = generateId();
-
-      dispatch({
-        type: 'UPDATE_LAYER',
-        datasourceId: state.activeDatasourceId!,
-        layerId: newLayerId,
-        updater: props.datasourceMap[state.activeDatasourceId!].insertLayer,
-      });
+  const framePublicAPI: FramePublicAPI = useMemo(
```

Comment:
Probably a `ref` here would be even better, but it requires way more refactoring than just this place.